packages/malloy/src/model/malloy_query.ts

```diff
@@ -1406,6 +1400,66 @@ class QueryField extends QueryNode {
     }
   }
 
+  generateAppliedFilter(
+    context: QueryStruct,
+    filterMatchExpr: FilterMatchExpr
+  ): string {
+    let filterExpr = filterMatchExpr.kids.filterExpr;
+    while (filterExpr.node === '()') {
+      filterExpr = filterExpr.e;
+    }
+    if (filterExpr.node === 'parameter') {
+      const name = filterExpr.path[0];
+      context.eventStream?.emit('source-argument-compiled', {name});
+      const argument = context.arguments()[name];
+      if (argument.value) {
+        filterExpr = argument.value;
+      } else {
+        throw new Error(
+          `Parameter ${name} was expected to be a filter expression`
+        );
+      }
+    }
+    if (filterExpr.node !== 'filterLiteral') {
+      throw new Error(
+        'Can only use filter expression literals or parameters as filter expressions'
+      );
+    }
+    const filterSrc = filterExpr.filterSrc;
+    let fParse: FilterParserResponse<FilterExpression>;
+    switch (filterMatchExpr.dataType) {
+      case 'string':
+        fParse = StringFilterExpression.parse(filterSrc);
+        break;
+      case 'number':
+        fParse = NumberFilterExpression.parse(filterSrc);
+        break;
+      case 'boolean':
+        if (context.dialect.booleanAsNumbers) {
+          throw new Error(
+            'Boolean filter expression not supported on this connection type'
+          );
+        }
```

Comment:
Is this just a temporary thing where eventually we will compile boolean filter expressions as number comparisons in SQL?

Reply:
Someone who cares about filter expressions could. The will to live drained out of me at this point. Boolean filter expressions, which are stupid anyway, don't work on MySQL which is stupid also. It's a that I am proposing is part of the MVP of fitlers.

Reply:
The problem is, when I read a schema, the type will be number, so even if I have thought about that value as a boolean forever, I can't apply a boolean filter to it uniess it is a local dimension to the input source I am processing. You would have to write 

```
dimension: hotOrCold is isCold::boolean ~ f'true' or isHot::boolean ~ f'true'
```

if isCold and isHot came from a table

Reply:
better to just fail hard and make them write

```
dimension: hotOrCol is isHot or isCold`
```

Reply:
One of the many reasons explorer should be writing stable lambdas not filter expressions

Reply:
I agree that the case where you are reading from the schema is problematic... But there are also cases where you define dimensions which are booleans, that you would want to use filter expressions on.

```
dimension: x is distance > 1000
dimension: filterthing is x ~ f`true`
```

Though I agree boolean filter expressions are not that useful... fine to leave it for now...

I also wonder whether someday we'll support:

```
source: s is conn.table('foo') include {
  some_integer_field::boolean
}
```
and sorta auto-cast `some_integer_field` to a boolean handwave handwave